src/main/java/lanat/parsing/errors/ParseStateErrorBase.java

```diff
@@ -83,29 +86,17 @@ public ParseStateErrorBase(@NotNull T errorType, int tokenIndex) {
 		}
 	}
 
+	@SuppressWarnings("unchecked")
 	private @NotNull List<@NotNull MMethod<?>> getAnnotatedMethods() {
-		/* todo: replace with this in the next mirror update
-		return mirror(this.getClass()).getSuperClassUntil(clazz -> clazz.getMethodCount() > 0)
-			.getMethods().stream()
-			.filter(method -> method.isAnnotatedWith(Handler.class))
-			.toList();
-		*/
-
-		List<MMethod<?>> methods;
-		MClass<?> currentClass = mirror(this.getClass());
-
-		// if there are no methods defined, get super class
-		// this is done for cases like usage of anonymous classes
-		while ((methods = currentClass.getMethods().toList()).size() == 0)
-			currentClass = mirror(currentClass.getRawClass().getSuperclass());
-
-		return methods.stream().filter(method -> method.isAnnotatedWith(Handler.class)).toList();
+		return mirror(this.getClass())
+			.getSuperclassUntilIncludingSelf(MClass::hasMethods)
+			.<List<MMethod<?>>>map(objectMClass -> objectMClass.getMethods(Filter.forMethods().withAnnotations(Handler.class))
+			.collect(Collectors.toList()))
+			.orElseGet(List::of);
 	}
 
 	private boolean isHandlerMethod(@NotNull MMethod<?> method, @NotNull String handlerName) {
-		return method.getAnnotationOfType(Handler.class)
-			.map(handler -> handler.value().equals(handlerName))
-			.orElse(false);
+		return method.getAnnotationOfType(Handler.class).map(handler -> handler.value().equals(handlerName)).orElse(false);
```

Comment:
Should this be split up into multiple lines?
```suggestion
		return method.getAnnotationOfType(Handler.class)
				.map(handler -> handler.value().equals(handlerName))
				.orElse(false);
```